internal/database/drivers/drivers.go

```diff
@@ -25,12 +25,16 @@ import (
 )
 
 type DatabaseDriver interface {
-	Get(*url.URL, *log.Entry) (DatabaseDriver, error)
-	SupportsScheme(string) bool
-	Connect(context.Context) error
 	GetEvents(context.Context) ([]schema.EiffelEvent, error)
 	SearchEvent(context.Context, string) (schema.EiffelEvent, error)
 	UpstreamDownstreamSearch(context.Context, string) ([]schema.EiffelEvent, error)
 	GetEventByID(context.Context, string) (schema.EiffelEvent, error)
+}
+
+type Database interface {
+	Get(*url.URL, *log.Entry) (Database, error)
+	Driver() (DatabaseDriver, error)
+	Connect(context.Context) error
+	SupportsScheme(string) bool
 	Close(context.Context) error
```

Comment:
Isn't this naming backwards? Having a Database from which you obtain a driver for performing the DB operations seems odd. The other way around makes more sense to me.

Secondly, I'm not sure I like the idea of having an uninitialized Database interface pointer in the

    var databases = []drivers.Database{&mongodb.Database{}}

slice. Only the Get and SupportScheme methods make sense in that context, and attempting to call any other method on `databases[0]` will result in a panic. I think we'd get a better API if the type we're listing in the  `databases` slice only has two methods, Get and SupportScheme, and that Get would return a reference to another interface, which could either be the interface that implements all the operations (GetEvents et al) or an interface with Connect and Close methods (and Connect returning an interface pointer with GetEvents and friends). In the former case the Connect method could be optional to call, with the operation methods calling it implicitly if a connection hasn't already been established.

Reply:
Made a few changes due to this comment. Firstly I changed the naming, thank you!

The drivers that are now in the "databases" slice only have Get() and SupportsScheme() so that the exported API is usable.
Changed so that Get() returns a Database that is connected to mongodb.
Had to update tests and added some helper functions to mock the database drivers.